LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/OTAAHelper/OTAAKeysGenerator.cs

```diff
@@ -10,31 +10,30 @@ namespace LoRaTools
 
     public static class OTAAKeysGenerator
     {
-        public static DevAddr GetNwkId(uint netId)
+        public static DevAddr GetNwkId(NetId netId)
         {
             var address = RandomNumberGenerator.GetInt32(toExclusive: DevAddr.MaxNetworkAddress + 1);
             // The 7 LBS of the NetID become the NwkID of a DevAddr:
-            return new DevAddr(unchecked((byte)netId) & 0b0111_1111, address);
+            return new DevAddr(unchecked((byte)netId.NetworkId), address);
         }
 
-        public static NetworkSessionKey CalculateNetworkSessionKey(byte[] type, byte[] appnonce, byte[] netid, DevNonce devNonce, AppKey appKey)
+        public static NetworkSessionKey CalculateNetworkSessionKey(byte[] type, byte[] appnonce, NetId netid, DevNonce devNonce, AppKey appKey)
         {
             var keyString = CalculateKey(type, appnonce, netid, devNonce, appKey);
             return NetworkSessionKey.Parse(keyString);
         }
 
-        public static AppSessionKey CalculateAppSessionKey(byte[] type, byte[] appnonce, byte[] netid, DevNonce devNonce, AppKey appKey)
+        public static AppSessionKey CalculateAppSessionKey(byte[] type, byte[] appnonce, NetId netid, DevNonce devNonce, AppKey appKey)
         {
             var keyString = CalculateKey(type, appnonce, netid, devNonce, appKey);
             return AppSessionKey.Parse(keyString);
         }
 
         // don't work with CFLIST atm
-        private static string CalculateKey(byte[] type, byte[] appnonce, byte[] netid, DevNonce devNonce, AppKey appKey)
+        public static string CalculateKey(byte[] type, byte[] appnonce, NetId netid, DevNonce devNonce, AppKey appKey)
```

Comment:
Any reason to change the accessibility of this method? It doesn't look like it's being used beyond this class with this PR.

Reply:
Also, consider renaming parameter `netid` or `netId` that would be more conventional.